src/wp-includes/fonts/class-wp-font-collection.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Font Collection class.
+ *
+ * This file contains the Font Collection class definition.
+ *
+ * @package    WordPress
+ * @subpackage Font Library
+ * @since      6.4.0
+ */
+
+/**
+ * Font Collection class.
+ *
+ * @since 6.4.0
+ */
+class WP_Font_Collection {
+
+	/**
+	 * Font collection configuration.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @var array
+	 */
+	private $config;
+
+	/**
+	 * WP_Font_Collection constructor.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array $config Font collection config options.
+	 *                      See {@see wp_register_font_collection()} for the supported fields.
+	 * @throws Exception If the required parameters are missing.
+	 */
+	public function __construct( $config ) {
+		if ( empty( $config ) || ! is_array( $config ) ) {
+			throw new Exception( 'Font Collection config options is required as a non-empty array.' );
+		}
+
+		if ( empty( $config['id'] ) || ! is_string( $config['id'] ) ) {
+			throw new Exception( 'Font Collection config ID is required as a non-empty string.' );
+		}
+
+		if ( empty( $config['name'] ) || ! is_string( $config['name'] ) ) {
+			throw new Exception( 'Font Collection config name is required as a non-empty string.' );
+		}
+
+		if ( empty( $config['src'] ) || ! is_string( $config['src'] ) ) {
+			throw new Exception( 'Font Collection config "src" option is required as a non-empty string.' );
+		}
+
+		$this->config = $config;
+	}
+
+	/**
+	 * Gets the font collection config.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return array An array containing the font collection config.
+	 */
+	public function get_config() {
+		return $this->config;
+	}
+
+	/**
+	 * Gets the font collection data.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return array|WP_Error An array containing the list of font families in theme.json format on success,
+	 *                        else an instance of WP_Error on failure.
+	 */
+	public function get_data() {
+		// If the src is a URL, fetch the data from the URL.
+		if ( false !== strpos( $this->config['src'], 'http' ) && false !== strpos( $this->config['src'], '://' ) ) {
```

Comment:
Could we use `str_contains`

Reply:
See https://core.trac.wordpress.org/ticket/49652 and https://core.trac.wordpress.org/ticket/58206

Reply:
good to know. Done

Reply:
A polyfill was added into Core for it in WP 5.9. Handy and more readable IMO

Reply:
Noting that this implementation for font collections has changed, but we are checking to use `str_contains` rather than `strpos`, where appropriate. See https://github.com/WordPress/gutenberg/pull/58675